### DISCUSSION TITLE
How to query the message status by snsClient.publish returned messageid

### DISCUSSION THREAD

**Discussion by asd435658185** [aws/aws-sdk-java#2789] [Q&A] [answered]
### Describe the feature

SnsClient.publish
Invalid phone nunmber calls publish,and the return vallue is always 200(successfull).
How to query the message status by snsClient.publish returned messageid

### Use Case

snsClient.publish(PublishRequest.builder().message("xxxxx").phoneNumber("xxxxxxxx").build).
How to query the message status by snsClient.publish returned messageid

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### AWS Java SDK version used

2.15.0

### JDK version used

1.8

### Operating System and version

window10

**Comment by debora-ito** [comment, marked as answer]
Hi @asd435658185 

SNS messages get published to a Topic, and to receive the messages you must subscribe something to the Topic, for example an SQS queue, a Lambda function, an email, etc. For more info check https://docs.aws.amazon.com/sns/latest/dg/sns-create-subscribe-endpoint-to-topic.html
